client/gui/WindowHandler.h

```diff
@@ -17,9 +17,6 @@ class WindowHandler
 	/// (includes adventure map, window windows, all kind of active dialogs, and so on)
 	std::vector<std::shared_ptr<IShowActivatable>> windowsStack;
 
-	/// Temporary list of recently popped windows
-	std::vector<std::shared_ptr<IShowActivatable>> disposed;
-
```

Comment:
While I am not against it, this could be potentially dangerous change.

Please check scenario when:
- you build something in town. I think we recreate castle interface when this happens.
- you have multiple towns of the same type - switching between them should not break music playback (or crash obviously)
- you have multiple towns of different types - music should change correctly

Basically - make sure that `CCastleInterface::~CCastleInterface` is called at the right moment & has correct logic. Can't think of any other similar cases at the moment.

Reply:
Yes, there were issues with the music. Removed unnecessary changes.